src/webui/src/components/trial-detail/TableList.tsx

```diff
@@ -45,749 +39,460 @@ echarts.registerTheme('my_theme', {
     color: '#3c8dbc'
 });
 
-const horizontalGapStackTokens: IStackTokens = {
-    childrenGap: 20,
-    padding: 10
+type SearchOptionType = 'id' | 'trialnum' | 'status' | 'parameters';
+const searchOptionLiterals = {
+    id: 'ID',
+    trialnum: 'Trial No.',
+    status: 'Status',
+    parameters: 'Parameters'
 };
 
-interface TableListProps {
-    pageSize: number;
-    tableSource: Array<TableRecord>;
-    columnList: string[]; // user select columnKeys
-    changeColumn: (val: string[]) => void;
-    trialsUpdateBroadcast: number;
+const defaultDisplayedColumns = ['sequenceId', 'id', 'startTime', 'endTime', 'duration', 'status', 'metric/default'];
+
+function _copyAndSort<T>(items: T[], columnKey: string, isSortedDescending?: boolean): any {
+    const key = columnKey as keyof T;
+    return items.slice(0).sort(function(a: T, b: T): any {
+        if (a[key] === undefined) {
+            return 1;
+        }
+        if (b[key] === undefined) {
+            return -1;
```

Comment:
because default metric column may be NaN | Infinity or others, so you should change it back. [example](https://github.com/microsoft/nni/blob/master/src/webui/src/components/trial-detail/TableList.tsx#L582)

Reply:
Updated and refactored this part. Please check.

Reply:
Please check the number column sort: duration and default metric.